scripts/prep-release.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+# Helper script to prepare a release for the Go SDK.
+
+# Read the version number and build number from the respective files
+current_version_number=$(< internal/version)
+current_build_number=$(< internal/version-build)
+
+version_file="internal/version"
+build_file="internal/version-build"
+
+# Function to validate the version number format x.y.z(-beta.w)
+validate_and_update_version_number() {
+
+    while true; do
+        # Prompt the user to input the version number
+        read -p "Enter the version number (format: x.y.z(-beta.w)): " version
+
+        # Validate the version number format
+        if [[ "${version}" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-beta\.[0-9]+)?$ ]]; then        
+            # Write the valid version number to the file
+            echo "${version}" > "${version_file}"
+            echo "New version number is: ${version}"
+            return 0
+        else
+            echo "Invalid version number format: ${version}"
+            echo "Please enter a version number in the 'x.y.z(-beta.w)' format."
+        fi
+    done
+}
+
+# Function to validate the build number format.
+# SEMVER Format: Mmmppbb - 7 Digits 
+update_and_validate_build_number() {
+    while true; do
+        # Prompt the user to input the build number
+        read -p "Enter the build number (format: Mmmppbb): " build
+
+        # Validate the build number format
+        if [[ "${build}" =~ ^[0-9]{7}$ ]]; then
+            # Write the valid build number to the file
+            echo "${build}" > "${build_file}"
+            echo "New build number is: ${build}"
+            return 0
+        else
+            echo "Invalid build number format: ${build}"
+            echo "Please enter a build number in the 'Mmmppbb' format."
+        fi
+    done
+}
+
+# Update and Validate the version number
+validate_and_update_version_number
+
+# Update and Validate the build number
+update_and_validate_build_number 
```

Comment:
Let's name these both `update_and_validate` - or anything else consistent.

Reply:
Removed the number from both functions, but as one function updates and validates the version and the other does the build.

Reply:
My point is that one is called  "update and validate" and the other "validate and update".

Reply:
Ohh.. I didn't realize lol fixed.